async_service/trio.py

```diff
@@ -188,6 +188,10 @@ def run_task(
         daemon: bool = False,
         name: str = None,
     ) -> None:
+        if not self.is_running or self.is_cancelled:
```

Comment:
Murr, I realize that `manager.is_running` is broken since it should definitely not return true if `is_cancelled` is true.  Need to fix this.